diff -uNr pilrc/font.c pilrc.new/font.c
--- pilrc/font.c	Fri Feb  4 06:54:08 2000
+++ pilrc.new/font.c	Mon Feb 28 02:40:05 2000
@@ -95,6 +95,48 @@
 	}
 
 /*-----------------------------------------------------------------------------
+|	IsKoreanHanme
+|	
+|		Check the double byte char is Korean coded(Large Font, for HanMe)
+-----------------------------------------------------------------------------*/
+int IsKoreanHanme(unsigned char *cp, int *pdxChar)
+{
+
+/* Korean code rule (by JaeMok Jeong)
+	first  byte range 0xb0..0xc8             (high byte)
+	second byte range 0xa1..0xfe             (low  byte)
+*/
+	if ((*cp>= 0xb0 && *cp<=0xc8) && (*(cp+1)>=0xa1 && *(cp+1)<=0xfe)) {
+		*pdxChar = 11;		/* not sure about this, hanme font width */
+		return 2;
+	}
+
+	return 0;
+
+}
+
+/*-----------------------------------------------------------------------------
+|	IsKoreanHantip
+|	
+|		Check the double byte char is Korean coded(Small Font, for Hantiip)
+-----------------------------------------------------------------------------*/
+int IsKoreanHantip(unsigned char *cp, int *pdxChar)
+	{
+
+/* Korean code rule (by JaeMok Jeong)
+	first  byte range 0xb0..0xc8             (high byte)
+	second byte range 0xa1..0xfe             (low  byte)
+*/
+	if ((*cp>= 0xb0 && *cp<=0xc8) && (*(cp+1)>=0xa1 && *(cp+1)<=0xfe))
+		{
+		*pdxChar = 8;		/* hantip font width = 8 */
+		return 2;
+		}
+	return 0;
+
+	}
+
+/*-----------------------------------------------------------------------------
 |	IsJapanese
 |	
 |		Check the double byte char is Japanese coded
@@ -747,6 +789,12 @@
 			SetBuiltinFont(0, font0Hebrew, 0xe0, sizeof(font0Hebrew));
 			SetBuiltinFont(1, font1Hebrew, 0xe0, sizeof(font1Hebrew));
 			SetBuiltinFont(2, font2Hebrew, 0xe0, sizeof(font2Hebrew));
+			break;
+		case fontKoreanHanme:
+			pfnChkCode = IsKoreanHanme;
+			break;
+		case fontKoreanHantip:
+			pfnChkCode = IsKoreanHantip;
 			break;
 		}
 
diff -uNr pilrc/font.h pilrc.new/font.h
--- pilrc/font.h	Tue Apr 13 13:04:02 1999
+++ pilrc.new/font.h	Mon Feb 28 02:12:27 2000
@@ -18,5 +18,7 @@
 #define fontJapanese 2	
 #define fontChineseBig5 3
 #define fontChineseGB 4
+#define fontKoreanHanme 5	
+#define fontKoreanHantip 6	
 
 #endif /*__FONT_H__ */
diff -uNr pilrc/main.c pilrc.new/main.c
--- pilrc/main.c	Mon Feb  7 22:56:48 2000
+++ pilrc.new/main.c	Mon Feb 28 02:39:47 2000
@@ -50,6 +50,8 @@
         "                     The current directory is always searched\n"
         "        -R <resfile> Generate JUMP/PilA .res file\n"
         "        -H <incfile> Autoassign IDs and write .h file with #defines\n"
+        "        -Fkt         Use Korean font widths(hantip font)\n"
+        "        -Fkm         Use Korean font widths(hanme font)\n"
         "        -F5          Use Big5 Chinese font widths\n"
         "        -Fg          Use GB Chinese font widths\n"
         "        -Fj          Use Japense font widths\n"
@@ -137,6 +139,19 @@
     // be quiet?
     if (FSzEqI(rgszArg[i], "-q")) {
       vfQuiet = fTrue;
+      continue;
+    }
+			
+    // font korean(by jmjeong@oopsla.snu.ac.kr, 2/26/00)
+	//	 not accurate for hanme(font information must be refined)
+	//
+    if (FSzEqI(rgszArg[i], "-Fkm")) {
+      fontType = fontKoreanHanme;
+      continue;
+    }
+	//
+    if (FSzEqI(rgszArg[i], "-Fkt")) {
+      fontType = fontKoreanHantip;
       continue;
     }
 			
